src/graph/create-entity.ts

```diff
@@ -0,0 +1,75 @@
+import { publishEdit } from '../ipfs.js';
+import { Triple } from '../triple.js';
+import type { SetTripleOp } from '../types.js';
+import { DEFAULT_API_HOST } from './constants.js';
+
+type Params = {
+  id: string;
+  spaceId: string;
+  accountId: string;
+  data: {
+    [filedName: string]: string;
+  };
+  mapping: {
+    [filedName: string]: string;
+  };
+  options?: {
+    apiHost?: string;
+    network?: 'TESTNET' | 'MAINNET';
+  };
+};
+
+export const createEntity = async ({ id, data, spaceId, accountId, mapping, options }: Params) => {
+  const ops: SetTripleOp[] = [];
+  const apiHost = options?.apiHost || DEFAULT_API_HOST;
+  const network = options?.network || 'TESTNET';
+  for (const [fieldName, value] of Object.entries(data)) {
+    const attributeId = mapping[fieldName];
+    if (!attributeId) {
+      throw new Error(`Attribute ID for field ${fieldName} not found`);
+    }
+    let attributeValueType: 'TEXT' | 'CHECKBOX' | 'NUMBER' | 'URL' | 'POINT';
+    switch (typeof value) {
+      case 'string':
+        attributeValueType = 'TEXT';
+        break;
+      case 'boolean':
+        attributeValueType = 'CHECKBOX';
+        break;
+      case 'number':
+        attributeValueType = 'NUMBER';
+        break;
+      default:
+        throw new Error('Not implemented value type');
+    }
```

Comment:
So a couple points
- We probably want to make sure we support creating relations before we introduce something like this since creating relations is _very_ common
- I think we should use the attribute id's value type as the source of truth for creating an entity rather than the JS value's primitive type. This would require fetching the data for each attribute and comparing it. But then we need to think about spaces and this starts to get more complex and verge into hypergraph territory. WDYT?

Reply:
Yeah I think this is where it's tricky since the op itself is the source of truth for what the value type is, and I don't think at this low level we want to force and validation or anything since developers should be able to write whatever they want to the KG.